test/test_named_destinations.py

```diff
@@ -0,0 +1,196 @@
+from pathlib import Path
+
+from fpdf import FPDF
+from fpdf.enums import XPos, YPos
+from test.conftest import assert_pdf_equal
+
+import pytest
+
+HERE = Path(__file__).resolve().parent
+
+
+def test_named_destinations(tmp_path):
+    """Test PDF generation with named destinations and links.
+
+    This test demonstrates how to create a hierarchical structure of named destinations
+    by reserving names first and then binding them to pages later.
+    """
+    # Create PDF
+    pdf = FPDF()
+
+    # Create the first page first
+    pdf.add_page()
+
+    # Then create all named destinations
+    # We need to specify the page number for each destination
+    main_toc_link = pdf.add_link(name="dests/main", page=1)
+    sub1_link = pdf.add_link(name="dests/subsection1", page=2)
+    sub2_link = pdf.add_link(name="dests/subsection2", page=3)
```

Comment:
```suggestion
```

With the suggested change you don't need to reserved or ever keeping reference to the dest/subsection2 link - it's a pure named destination